cla_backend/apps/cla_eventlog/tests/test_find_and_delete_old_cases.py

```diff
@@ -0,0 +1,233 @@
+        self.assertEqual(EODDetails.objects.count(), 0)
+        self.assertEqual(EligibilityCheck.objects.count(), 0)
+        self.assertEqual(DiagnosisTraversal.objects.count(), 0)
+        self.assertEqual(PersonalDetails.objects.count(), 0)
+
+    def test_old_case_with_old_means_test_changed_is_deleted(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        case = self.create_case(date, "legalaid.eligible_case")
+        self.create_event_log_for_case(case, "MT_CHANGED", _make_datetime(year=2014, month=5, day=30, hour=9))
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 1)
+
+        self.delete_old_cases(dt)
+        self.assertEqual(Case.objects.count(), 0)
+        self.assertEqual(Log.objects.count(), 0)
+        self.assertEqual(AuditLog.objects.count(), 0)
+        self.assertEqual(Complaint.objects.count(), 0)
+        self.assertEqual(EODDetails.objects.count(), 0)
+        self.assertEqual(EligibilityCheck.objects.count(), 0)
+        self.assertEqual(DiagnosisTraversal.objects.count(), 0)
+        self.assertEqual(PersonalDetails.objects.count(), 0)
+
+    def test_old_and_new_cases_are_deleted_correctly(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        case_0 = self.create_case(date, "legalaid.eligible_case")
+        case_1 = self.create_case(date, "legalaid.eligible_case")
+        case_2 = self.create_case(date, "legalaid.eligible_case")
+        self.create_event_log_for_case(case_0, "MT_CHANGED", _make_datetime(year=2014, month=5, day=30, hour=9))
+        self.create_event_log_for_case(case_0, "MT_CHANGED", _make_datetime(year=2018, month=5, day=30, hour=9))
+        self.create_event_log_for_case(case_1, "MT_CHANGED", _make_datetime(year=2014, month=5, day=30, hour=9))
+        self.create_event_log_for_case(case_2, "MT_CHANGED", _make_datetime(year=2020, month=4, day=27, hour=9))
+
+        audit_log_1 = make_recipe("cla_auditlog.audit_log")
+        audit_log_2 = make_recipe("cla_auditlog.audit_log")
+        audit_log_3 = make_recipe("cla_auditlog.audit_log")
+        eod_0 = make_recipe("legalaid.eod_details", case=case_0)
+        eod_1 = make_recipe("legalaid.eod_details", case=case_1)
+        eod_2 = make_recipe("legalaid.eod_details", case=case_2)
+        make_recipe("complaints.complaint", eod=eod_0, audit_log=[audit_log_1])
+        make_recipe("complaints.complaint", eod=eod_1, audit_log=[audit_log_2])
+        make_recipe("complaints.complaint", eod=eod_2, audit_log=[audit_log_3])
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 2)
+
+        self.delete_old_cases(dt)
+        valid_case = Case.objects.filter(id=case_2.id)
+        self.assertEqual(valid_case.count(), 1)
```

Comment:
I would also assert that the total number of cases is 1, but I like that you are asserting it's the right one. But this assertion alone would pass if both cases still existed.

Ooh, even better, you could do something like
`self.assertEqual(Case.objects.values_list('id', flat=True), [case_2.id])`

This will do both things at once!

Reply:
Ahh yeah true. Awesome, I change `Case.objects.values_list('id', flat=True)` to `list(Case.objects.values_list('id', flat=True))` as the first one is a queryset.

Reply:
That's really great work